compiler/rustc_lint/src/levels.rs

```diff
@@ -123,17 +123,19 @@ fn lints_that_dont_need_to_run(tcx: TyCtxt<'_>, (): ()) -> FxIndexSet<LintId> {
     let dont_need_to_run: FxIndexSet<LintId> = store
         .get_lints()
         .into_iter()
+        .filter(|lint| {
+            // Lints that show up in future-compat reports must always be run.
+            let has_future_breakage =
+                lint.future_incompatible.is_some_and(|fut| fut.reason.has_future_breakage());
+            !has_future_breakage && !lint.eval_always
+        })
```

Comment:
Is there a reason to have these two as different methods?

Reply:
I think it's easier to first do the pure filtering, and then the `map` part that also transforms the data. This also reduces rightward drift in the `filter_map` closure.

Reply:
could the fitire breakage check be moved earlier? i.e. automatically set `eval_always` for all future breakage lints

Reply:
That would require putting the logic for this into the `declare_lint` macro. IMO that's a bad idea, I'd rather not have this logic hidden in a macro.